terraform/utils.go

```diff
@@ -326,3 +355,104 @@ func isTerraformPlan(content []byte) bool {
 
 	return hasResourceChanges && hasFormatVersion
 }
+
+// findBlockLinesFromJSON locates the start and end lines of a specific block or nested element within a block.
+// The file should contain structured data (e.g., JSON) and this function expects to search for blocks with specific names.
+func findBlockLinesFromJSON(file *os.File, blockName string, pathName ...string) (int, int) {
```

Comment:
Does the kics parser have any info about line numbers for state files/from the JSON parser?

Reply:
It does provide a line number of various properties, but that did not help to get the block start and end line. For example, below is a resource sample - 

The start and end line number for this resource is 29 - 127.
In this sample, there are line numbers available for various properties of the resource instances, but instances are a child property of the resource; there are other properties also available, like name, type and mode, for which line numbers are not there.
```
EXTRA_VALUE_AT_END="map[instances:[map[attributes:map[_kics_lines:map[_kics__default:{37 []} _kics_architecture:{38 []} _kics_arn:{39 []} _kics_block_device_mappings:{40 [map[_kics__default:{42 []} _kics_device_name:{42 []} _kics_ebs:{43 []} _kics_no_device:{52 []} _kics_virtual_name:{53 []}] map[_kics__default:{42 []} _kics_device_name:{56 []} _kics_ebs:{57 []} _kics_no_device:{58 []} _kics_virtual_name:{59 []}] map[_kics__default:{42 []} _kics_device_name:{62 []} _kics_ebs:{63 []} _kics_no_device:{64 []} _kics_virtual_name:{65 []}]]} _kics_boot_mode:{68 []} _kics_creation_date:{69 []} _kics_deprecation_time:{70 []} _kics_description:{71 []} _kics_ena_support:{72 []} _kics_executable_users:{73 []} _kics_filter:{74 [map[_kics__default:{76 []} _kics_name:{76 []} _kics_values:{77 [map[_kics__default:{78 []}]]}] map[_kics__default:{76 []} _kics_name:{76 []} _kics_values:{83 [map[_kics__default:{84 []}]]}]]} _kics_hypervisor:{88 []} _kics_id:{89 []} _kics_image_id:{90 []} _kics_image_location:{91 []} _kics_image_owner_alias:{92 []} _kics_image_type:{93 []} _kics_imds_support:{94 []} _kics_include_deprecated:{95 []} _kics_kernel_id:{96 []} _kics_most_recent:{97 []} _kics_name:{98 []} _kics_name_regex:{99 []} _kics_owner_id:{100 []} _kics_owners:{101 [map[_kics__default:{102 []}]]} _kics_platform:{104 []} _kics_platform_details:{105 []} _kics_product_codes:{106 []} _kics_public:{107 []} _kics_ramdisk_id:{108 []} _kics_root_device_name:{109 []} _kics_root_device_type:{110 []} _kics_root_snapshot_id:{111 []} _kics_sriov_net_support:{112 []} _kics_state:{113 []} _kics_state_reason:{114 []} _kics_tags:{118 []} _kics_timeouts:{119 []} _kics_tpm_support:{120 []} _kics_usage_operation:{121 []} _kics_virtualization_type:{122 []}] architecture:x86_64 arn:arn:aws:ec2:us-east-1::image/ami-055744c75048d8296 block_device_mappings:[map[device_name:/dev/sda1 ebs:map[_kics_lines:map[_kics__default:{43 []} _kics_delete_on_termination:{44 []} _kics_encrypted:{45 []} _kics_iops:{46 []} _kics_snapshot_id:{47 []} _kics_throughput:{48 []} _kics_volume_size:{49 []} _kics_volume_type:{50 []}] delete_on_termination:true encrypted:false iops:0 snapshot_id:snap-0a03cace5a5d32a40 throughput:0 volume_size:8 volume_type:gp2] no_device: virtual_name:] map[device_name:/dev/sdb ebs:map[_kics_lines:map[_kics__default:{57 []}]] no_device: virtual_name:ephemeral0] map[device_name:/dev/sdc ebs:map[_kics_lines:map[_kics__default:{63 []}]] no_device: virtual_name:ephemeral1]] boot_mode:legacy-bios creation_date:2023-05-31T17:00:13.000Z deprecation_time:2025-05-31T17:00:13.000Z description:Canonical, Ubuntu, 18.04 LTS, amd64 bionic image build on 2023-05-31 ena_support:true executable_users:<nil> filter:[map[name:name values:[ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-*]] map[name:virtualization-type values:[hvm]]] hypervisor:xen id:ami-055744c75048d8296 image_id:ami-055744c75048d8296 image_location:amazon/ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-20230531 image_owner_alias:amazon image_type:machine imds_support: include_deprecated:false kernel_id: most_recent:true name:ubuntu/images/hvm-ssd/ubuntu-bionic-18.04-amd64-server-20230531 name_regex:<nil> owner_id:099720109477 owners:[099720109477] platform: platform_details:Linux/UNIX product_codes:[] public:true ramdisk_id: root_device_name:/dev/sda1 root_device_type:ebs root_snapshot_id:snap-0a03cace5a5d32a40 sriov_net_support:simple state:available state_reason:map[_kics_lines:map[_kics__default:{114 []} _kics_code:{115 []} _kics_message:{116 []}] code:UNSET message:UNSET] tags:map[_kics_lines:map[_kics__default:{118 []}]] timeouts:<nil> tpm_support: usage_operation:RunInstances virtualization_type:hvm] schema_version:0 sensitive_attributes:[]]] mode:data name:ubuntu provider:provider[\"registry.terraform.io/hashicorp/aws\"] type:aws_ami]"
```